spin/module.go

```diff
@@ -0,0 +1,229 @@
+	}
+
+	modules, err := resolveModules(canonicalTerragruntConfigPaths, terragruntOptions)
+	if err != nil {
+		return []*TerraformModule{}, err
+	}
+
+	externalDependencies, err := resolveExternalDependenciesForModules(canonicalTerragruntConfigPaths, modules, terragruntOptions)
+	if err != nil {
+		return []*TerraformModule{}, err
+	}
+
+	return crosslinkDependencies(mergeMaps(modules, externalDependencies), canonicalTerragruntConfigPaths)
+}
+
+// Go through each of the given .terragrunt config files and resolve the module that .terragrunt file represents it
+// into a TerraformModule struct. Note that this method will NOT fill in the Dependencies field of the TerraformModule
+// struct (see the crosslinkDependencies method for that). Return a map from module path to TerraformModule struct.
+func resolveModules(canonicalTerragruntConfigPaths []string, terragruntOptions *options.TerragruntOptions) (map[string]*TerraformModule, error) {
+	moduleMap := map[string]*TerraformModule{}
+
+	for _, terragruntConfigPath := range canonicalTerragruntConfigPaths {
+		module, err := resolveTerraformModule(terragruntConfigPath, terragruntOptions)
+		if err != nil {
+			return moduleMap, err
+		}
+		moduleMap[module.Path] = module
+	}
+
+	return moduleMap, nil
+}
+
+// Create a TerraformModule struct for the Terraform module specified by the given .terragrunt config file path. Note
+// that this method will NOT fill in the Dependencies field of the TerraformModule struct (see the
+// crosslinkDependencies method for that).
+func resolveTerraformModule(terragruntConfigPath string, terragruntOptions *options.TerragruntOptions) (*TerraformModule, error) {
+	modulePath, err := util.CanonicalPath(filepath.Dir(terragruntConfigPath), ".")
+	if err != nil {
+		return nil, err
+	}
+
+	opts := terragruntOptions.Clone(terragruntConfigPath)
+	terragruntConfig, err := config.ParseConfigFile(terragruntConfigPath, opts, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &TerraformModule{Path: modulePath, Config: *terragruntConfig, TerragruntOptions: opts}, nil
+}
+
+
+// Look through the dependencies of the modules in the given map and resolve the "external" dependency paths listed in
+// each modules config (i.e. those dependencies not in the given list of .terragrunt config canonical file paths).
+// These external dependencies are outside of the current working directory, which means they may not be part of the
+// environment the user is trying to spin-up or tear down. Therefore, this method also confirms whether the user wants
+// to actually apply those dependencies or just assume they are already applied. Note that this method will NOT fill in
+// the Dependencies field of the TerraformModule struct (see the crosslinkDependencies method for that).
+func resolveExternalDependenciesForModules(canonicalTerragruntConfigPaths []string, moduleMap map[string]*TerraformModule, terragruntOptions *options.TerragruntOptions) (map[string]*TerraformModule, error) {
```

Comment:
As it is now, yes, it'll prompt for each one. If that turns out to be problematic, we can explore improvements.